middleware/grpc/client.go

```diff
@@ -25,21 +20,25 @@ type clientHandler struct {
 // A ClientOption can be passed to NewClientHandler to customize the returned handler.
 type ClientOption func(*clientHandler)
 
-// A RPCHandler can be registered using WithRPCHandler to intercept calls to HandleRPC of a
-// handler for additional span customization.
-type RPCHandler func(span zipkin.Span, rpcStats stats.RPCStats)
-
-// WithRPCHandler allows one to add custom logic for handling a stats.RPCStats, e.g.,
+// WithClientRPCHandler allows one to add custom logic for handling a stats.RPCStats, e.g.,
 // to add additional tags.
-func WithRPCHandler(handler RPCHandler) ClientOption {
+func WithClientRPCHandler(handler RPCHandler) ClientOption {
 	return func(c *clientHandler) {
 		c.rpcHandlers = append(c.rpcHandlers, handler)
 	}
 }
 
+// WithClientRemoteServiceName will set the value for the remote endpoint's service name on
+// all spans.
+func WithClientRemoteServiceName(name string) ClientOption {
```

Comment:
if this is in client.go do we need client prefix? not sure of go idioms. In the text I would say something that this is the name of the server :P (or withServerName might work)

Reply:
It's in the same package as the server so can't have the same function appear twice. 

That said I'm not happy with it from an API usage / naming point either. Will have to inspect some more..

Reply:
Yeah I think server remote name is inconceivable, just erred on the side of consistency. But will remove it.